go/worker/keymanager/worker.go

```diff
@@ -302,12 +291,22 @@ func (w *Worker) updateStatus(status *api.Status, startedEvent *host.StartedEven
 
 	// Register as we are now ready to handle requests.
 	initOk = true
-	w.roleProvider.SetAvailable(func(n *node.Node) error {
+	w.roleProvider.SetAvailableWithCallback(func(n *node.Node) error {
 		rt := n.AddOrUpdateRuntime(w.runtime.ID())
 		rt.Version = startedEvent.Version
 		rt.ExtraInfo = cbor.Marshal(signedInitResp)
 		rt.Capabilities.TEE = startedEvent.CapabilityTEE
 		return nil
+	}, func(context.Context) error {
+		w.logger.Info("Key manager registered")
+
+		// Signal that we are initialized.
```

Comment:
Maybe mention here or where `initCh` is declared that KeyManager triggers `Initialized()` when it is ready to serve requests + is registered (in contrast to other workers where it's not mandatory to be registered).

Reply:
Good idea, changed the documentation comment for the `Initialized` method.